lib/cli/src/Cardano/CLI.hs

```diff
@@ -431,6 +449,79 @@ instance FromText CliWalletStyle where
 instance ToText CliWalletStyle where
     toText = toTextFromBoundedEnum SnakeLowerCase
 
+newtype DerivationPath = DerivationPath [DerivationIndex]
+    deriving (Show, Eq)
+
+instance FromText DerivationPath where
+    fromText x = DerivationPath <$> mapM fromText (T.splitOn "/" x)
+
+instance ToText DerivationPath where
+    toText (DerivationPath xs) = T.intercalate "/" $ map toText xs
+
+newtype DerivationIndex = DerivationIndex { unDerivationIndex :: Word32 }
+    deriving (Show, Eq)
+    deriving newtype (Bounded, Enum)
+
+firstHardenedIndex :: Word32
+firstHardenedIndex = getIndex $ minBound @(Index 'Hardened 'AddressK)
+
+instance FromText DerivationIndex where
+    fromText "" = Left $ TextDecodingError
+        "An empty string is not a derivation index!"
+    fromText x = do
+       -- NOTE: T.takeEnd will not throw, but may return "".
+       if T.takeEnd 1 x == "\'"
```

Comment:
I put a comment on ADP-81 about this. I don't think using the `'` character will work very well in the most common shells (sh, bash, zsh).

Reply:
I think the risk of turning `0'/0'` into `0/0` by accident makes this a valid concern. I changed to `H` as suggested by Matthias.